### PR TITLE
webui: add support for ISCSI storage for installation destination

### PR DESCRIPTION
**Pull request by KKoukiou**
* [x] Blocked on: https://github.com/rhinstaller/anaconda/pull/5281
* [x] Needs more test coverage
* [ ] Needs reverse chap support in the UI (followup)
* [ ] Needs better presentation of the error messages - now presenting directly what we get from backend - (followup)
* [ ] Needs designer review - the current design is self inspired (followup) - @garrett is in sick leave

https://github.com/rhinstaller/anaconda/assets/14921356/91385ff7-c150-49cc-a46f-08eee9fd1d9b

### CODE REVIEW

File: ui/webui/src/helpers/utils.js [hunk 1]
```diff
@@ -38,3 +40,19 @@ export const checkIfArraysAreEqual = (array1, array2) => {
...
+ */
+export const objectFromDBus = (dbusObject) => {
+    return Object.keys(dbusObject).reduce((acc, k) => { acc[k] = dbusObject[k].v; return acc }, {});
+};
+
+/* Converts an object with normal values to an object with variant string values
+ * @param {Object} object
+ * @returns {Object} The converted object
+ */
+export const objectToDBus = (object) => {
```
Comment (rvykydal) [inline ui/webui/src/helpers/utils.js:56]: Maybe we should indicate in the name that the function produces string variants.

**Review by vojtechtrefny — COMMENTED** [review]
Looks good to me in general.

One note: In Gtk UI the last "step" of ISCSI login shows all disks/targets that are now available and user can choose which ones will be added to the disk selection. With the WebUI the targets are shown but then on the Destination page "nothing happens" and it was a little bit confusing for me.

**Review by rvykydal — APPROVED** [review]
Looks great to me.

**Comment by KKoukiou** [comment]
> Looks good to me in general.
> 
> One note: In Gtk UI the last "step" of ISCSI login shows all disks/targets that are now available and user can choose which ones will be added to the disk selection. With the WebUI the targets are shown but then on the Destination page "nothing happens" and it was a little bit confusing for me.

I see your point. I guess the designer would have figured some better user experience. Maybe after closing the login dialog I can show a notification above the disk selector that now new disks are available, and mention the disk names.

**Comment by jstodola** [comment]
Katerina, can you also add an end2end test to cover the whole installation and boot process, please?

**Comment by KKoukiou** [comment]
> Katerina, can you also add an end2end test to cover the whole installation and boot process, please?

Sure. This checks that the iscsi device is selectable only, you 're right. I honestly did not make the test run the whole installation on the purpose, as believed that anaconda does not have any specific to iscsi code, after the device is selected. I see now that i was probably wrong, storage/bootloader/base.py for example has a lot of special cases for iscsi.

I will adjust the test to run full installation.

**Comment by KKoukiou** [comment]
https://github.com/rhinstaller/anaconda-webui/pull/31 moved.

**Pull request closed without merging**